design/attribute_test.go

```diff
@@ -145,3 +145,50 @@ func TestAttributeExprValidate(t *testing.T) {
 		}
 	}
 }
+
+func TestAttributeExprAllRequired(t *testing.T) {
+	cases := map[string]struct {
+		typ        DataType
+		validation *ValidationExpr
+		expected   []string
+	}{
+		"user type": {
+			typ: &UserTypeExpr{
+				AttributeExpr: &AttributeExpr{
+					Validation: &ValidationExpr{
+						Required: []string{"foo", "bar"},
+					},
+				},
+			},
+			expected: []string{"foo", "bar"},
+		},
+		"not user type": {
```

Comment:
I'm wondering if we need to test for this case? it works but having required attributes on a type that is not an object does not make a lot of sense.

Reply:
It makes sence. I remove the case and rename `user type` to `some required`.

Reply:
Fixed it. https://github.com/goadesign/goa/pull/1558/commits/581b2014316b876a5d61797b5bfc8bf0b7f8c7aa

Reply:
Thank you!